File: simple-stack-example-nestedstack/src/main/java/com/zhuinden/simplestackdemonestedstack/presentation/paths/main/mail/MailKey.java
package com.zhuinden.simplestackdemonestedstack.presentation.paths.main.mail;

import com.google.auto.value.AutoValue;
import com.zhuinden.simplestackdemonestedstack.R;
import com.zhuinden.simplestackdemonestedstack.application.Key;
import com.zhuinden.simplestackdemonestedstack.presentation.paths.main.MainView;

/**
 * Created by Zhuinden on 2017.02.19..
 */
@AutoValue
public abstract class MailKey
        extends Key {
    @Override
    public int layout() {
        return R.layout.path_mail;
    }

    public static MailKey create() {
        return new AutoValue_MailKey();
    }

    @Override
    public String stackIdentifier() {
        return MainView.StackType.MAIL.name();
    }
}

File: simple-stack-example-multistack-view/src/main/java/com/zhuinden/simplestackdemomultistack/features/main/mail/MailKey.kt
package com.zhuinden.simplestackdemomultistack.features.main.mail

import com.zhuinden.simplestackdemomultistack.R
import com.zhuinden.simplestackdemomultistack.application.MainActivity
import com.zhuinden.simplestackdemomultistack.core.navigation.MultistackKey
import kotlinx.android.parcel.Parcelize

@Parcelize
data class MailKey(private val placeholder: String = "") : MultistackKey() {
    override fun layout(): Int = R.layout.mail_view
    override fun stackIdentifier(): String = MainActivity.StackType.MAIL.name
}
